Sources/PackageGraph/PackageGraphRoot.swift

```diff
@@ -93,24 +98,39 @@ public struct PackageGraphRoot {
     public let dependencies: [PackageDependency]
 
     /// Create a package graph root.
-    public init(input: PackageGraphRootInput, manifests: [Manifest]) {
+    public init(input: PackageGraphRootInput, manifests: [Manifest], explicitProduct: String? = nil) {
         self.packageRefs = zip(input.packages, manifests).map { (path, manifest) in
             let identity = PackageReference.computeIdentity(packageURL: manifest.url)
             return PackageReference(identity: identity, path: path.pathString, kind: .root)
         }
         self.manifests = manifests
-        self.dependencies = input.dependencies
+
+        // FIXME: Special casing explicit products like this is necessary to pass the test suite and satisfy backwards compatibility. However, changing the dependencies based on the command line arguments may force pins to temporarily change, which can become a nuissance. Such pin switching can currently be worked around by declaring the executable product as a dependency of a dummy target. But in the future it might be worth providing a way of declaring them in the manifest without a dummy target, at which time the current special casing can be deprecated.
```

Comment:
Xcode doesn’t recognize any hard‐wrapped lines as part of the FIXME. Are you sure that’s what you want?

Reply:
The project has never adopted style rules, so I'd refer to @aciidb0mb3r on that. I just got in the habit of manually wrapping code and comment to never go past 120 characters. In terms of Xcode not recognizing a wrapped FIXME, one solution would be to split the comment in a short title and more verbose description:

```
// FIXME: Deprecate special casing once the manifest supports declaring used executable products.
// Special casing explicit products like this is necessary to pass the test suite and satisfy backwards compatibility. However, changing the dependencies based on the command line arguments may force pins to temporarily change, which can become a nuissance. Such pin switching can currently be worked around by declaring the executable product as a dependency of a dummy target. But in the future it might be worth providing a way of declaring them in the manifest without a dummy target, at which time the current special casing can be deprecated.
```

Reply:
I split out the summary into the fix‐me, and left the rest as comments like you suggested.